### PR TITLE
Kokoro rework

### PR DESCRIPTION
**Pull request by TheRoyalTnetennba**
* The config layout is pretty how the github checks will be organized.
* Currently the plan is one github check per gem for PR's. The gem will be loaded on a docker image with rbenv and the supported ruby_versions. The container will then run tests on each version sequentially. Will switch to running tests on each version in parallel as long as logs are accessible.
* Merges to master will create one separate github check per gem/ruby-version. This is to alleviate some of the ci issues surrounding longer running acceptance tests.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -903,10 +903,60 @@ task :compile do
...
+      File.open("./.kokoro/continuous/#{gem}-ruby-#{ruby_version}.cfg", 'w') do |f| 
+        f.write(kokoro_config(gem, ruby_version))
+      end
+    end
+
+  end
+end
+
+def kokoro_config gem, ruby_version = nil
+  lines = []
```
Comment (blowmage) [inline Rakefile]: Thoughts on using an ERB file instead of putting this inline?

Comment (TheRoyalTnetennba) [reply Rakefile]: I'm for it.

**Comment by TheRoyalTnetennba** [comment]
The Stackdriver ci task needs to build google-cloud-debugger before running.

**Comment by TheRoyalTnetennba** [comment]
Jobs are randomly timing out: 
1) https://source.cloud.google.com/results/invocations/93ff6dea-b1e2-4113-85f5-52cb0f9c7f4c/targets/cloud-devrel%2Fclient-libraries%2Fgoogle-cloud-ruby%2Fpresubmit%2Fgoogle-cloud/log

2) https://source.cloud.google.com/results/invocations/0b956f86-b7dd-49ba-a3cb-acb6bd52da49/targets/cloud-devrel%2Fclient-libraries%2Fgoogle-cloud-ruby%2Fpresubmit%2Fgoogle-cloud-tasks/log

3) https://source.cloud.google.com/results/invocations/939ec15a-c0d2-4070-9f2d-534ee4fa7781/targets/cloud-devrel%2Fclient-libraries%2Fgoogle-cloud-ruby%2Fpresubmit%2Fgoogle-cloud-env/log

File: stackdriver/Rakefile [hunk 2]
```diff
@@ -69,14 +69,16 @@ task :console do
...
+    @gcloud ||= Google::Cloud.new
+  end
 
   ARGV.clear
   IRB.start
 end
 
 desc "Generates JSON output from stackdriver .yardoc"
-task :jsondoc => :yard do
+task jsondoc: :yard do
```
Comment (blowmage) [inline stackdriver/Rakefile]: We just removed the `jsondoc` tasks. Please rebase this PR on master to get the latest.

Comment (blowmage) [reply stackdriver/Rakefile]: NM, I followed a notification that took me to a version of this PR before the commit of merging to master was added.

**Comment by blowmage** [comment]
I am unable to see the results of the builds. Every time I follow the "Details" link for a build I get a "Uh oh! Something went wrong." message.

**Comment by TheRoyalTnetennba** [comment]
@blowmage can you see if it was just a temporary hangup? I can't repro.

**Comment by dazuma** [comment]
I can see build results from my external account.

File: .kokoro/build.sh [hunk 3]
```diff
@@ -23,24 +27,44 @@ bundle update
...
+    (bundle update && bundle exec rake ci) || set_failed_status
+  done
   ;;
-post)
-  (bundle update && bundle exec rake circleci:post) || set_failed_status
+continuous)
+  cd $PACKAGE
+  (bundle update && bundle exec rake ci:acceptance) || set_failed_status
   ;;
 release)
```
Comment (blowmage) [inline .kokoro/build.sh]: Do we have release integration? I believe this is looking for the `CIRCLE_TAG` environment variable.

**Comment by blowmage** [comment]
Yes, it must have been temporary as those links are working now. But I was getting errors last night too. Oh well.

File: stackdriver/Rakefile [hunk 4]
```diff
@@ -116,7 +119,20 @@ namespace :ci do
   end
 end
 
-task :default => :test
+task default: :test
+
+desc "Compiles c extension for google-cloud-debugger"
+task :debugger_c do
+  header "compiling google-cloud-debugger", "*"
+  debugger_directory = Dir.getwd.split("/")[0...-1]
```
Comment (dazuma) [inline stackdriver/Rakefile]: Better to use methods of File to manipulate paths (so it works on windows).

```ruby
debugger_directory = File.join(File.dirname(Dir.getwd), "google-cloud-debugger")
Dir.chdir(debugger_directory) do
...
```

**Comment by blowmage** [comment]
So the solution now is to run each gem's tests for all available ruby versions in one job? So each gem gets its own checkmark? That will help with acceptance tests resource usage, and each ruby version isn't run concurrently.

**Comment by TheRoyalTnetennba** [comment]
@blowmage For PR's, yes. But they don't (at least currently) include acceptance tests. For merges to master, every gem will be run on every ruby version concurrently. So 108 different checks, each w/ acceptance tests. If resource usage is an issue, we can pare it down to only running acceptance tests on the 2.5.1 vm's.

File: .kokoro/build.sh (hunk 3 above)
Comment (TheRoyalTnetennba) [reply .kokoro/build.sh]: We don't fully yet. That's something I'll be working on once we have a functioning, satisfactory setup for PR's.

**Comment by blowmage** [comment]
Why are PRs and master configured differently? I thought the eventual plan was to run acceptance tests on PRs.

**Comment by blowmage** [comment]
> Are there security concerns with allowing acceptance tests to non-authorized users?

Yes.

**Comment by blowmage** [comment]
Ideally, we would like to test the different Ruby versions on the different platforms. So 36 gems * 3 ruby versions * 3 platforms = 324 jobs per build. And more once Ruby 2.6 is released.

**Comment by blowmage** [comment]
Will merging this PR make the other PRs start passing on Kokoro again?

**Comment by TheRoyalTnetennba** [comment]
Merging will resolve the kokoro failures on other PR's.

**Pull request merged**